src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/HaFlowExtended.groovy

```diff
@@ -156,9 +170,43 @@ class HaFlowExtended {
     }
 
     HaFlowRerouteResult reroute() {
+        log.debug("Reroute ha-flow '${haFlowId}'")
         northboundV2.rerouteHaFlow(haFlowId)
     }
 
+    HaFlowExtended retrieveDetails() {
+        log.debug("Getting ha-flow details '${haFlowId}'")
+        def haFlow = northboundV2.getHaFlow(haFlowId)
+        new HaFlowExtended(haFlow, northboundV2, topologyDefinition)
+    }
+
+    HaFlowExtended swap() {
+        log.debug("Swap ha-flow '${haFlowId}'")
+        def haFlow = northboundV2.swapHaFlowPaths(haFlowId)
+        new HaFlowExtended(haFlow, northboundV2, topologyDefinition)
+    }
+
+    HaFlowPingResult ping(HaFlowPingPayload pingPayload) {
```

Comment:
Let's accept `int` as parameter of this method. Anyway, it's only parameter which can be passed to HaFlowPingPayload. Also, some default value coudl be handy too.

Reply:
done